memgpt/main.py

```diff
@@ -438,6 +439,12 @@ async def main(
                         memgpt_agent.messages.pop()
                     continue
 
+                elif user_input.lower() == "/retry":
+                    print(f"Retry for another answer")
+                    memgpt_agent.messages.pop()

```

Comment:
@oderwat I think this code works if you assume the assistant always finishes with a send_message, since that's 2 messages (role assistant -> role function).

However in some cases the assistant might not finish with a function call, so we should probably modify the code to manage this case.

Maybe something like a "pop" that goes in a loop until the item about to be popped is role == user?

Reply:
I rewrote the whole implementation and this should not be a problem anymore